tests/trainer/test_checks.py

```diff
@@ -13,58 +13,40 @@
 )
 
 
-def test_error_on_no_train_step(tmpdir):
-    """ Test that an error is thrown when no `training_step()` is defined """
-    tutils.reset_seed()
-
-    class CurrentTestModel(LightningModule):
-        def forward(self, x):
-            pass
-
-    trainer_options = dict(default_root_dir=tmpdir, max_epochs=1)
-    trainer = Trainer(**trainer_options)
-
-    with pytest.raises(MisconfigurationException):
-        model = CurrentTestModel()
-        trainer.fit(model)
-
+# TODO: add matching messages
 
-def test_error_on_no_train_dataloader(tmpdir):
-    """ Test that an error is thrown when no `training_dataloader()` is defined """
+def test_wrong_train_setting(tmpdir):
+    """
+    * Test that an error is thrown when no `training_dataloader()` is defined
+    * Test that an error is thrown when no `training_step()` is defined
+    """
     tutils.reset_seed()
     hparams = tutils.get_default_hparams()
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
 
-    class CurrentTestModel(TestModelBase):
-        pass
-
-    trainer_options = dict(default_root_dir=tmpdir, max_epochs=1)
-    trainer = Trainer(**trainer_options)
+    with pytest.raises(MisconfigurationException):
+        model = EvalModelTemplate(hparams)
+        model.train_dataloader = LightningModule.train_dataloader
+        trainer.fit(model)
 
     with pytest.raises(MisconfigurationException):
-        model = CurrentTestModel(hparams)
+        model = EvalModelTemplate(hparams)
+        model.training_step = None
         trainer.fit(model)
 
 
-def test_error_on_no_configure_optimizers(tmpdir):
+def test_wrong_configure_optimizers(tmpdir):
     """ Test that an error is thrown when no `configure_optimizers()` is defined """
     tutils.reset_seed()
-
-    class CurrentTestModel(LightTrainDataloader, LightningModule):
-        def forward(self, x):
-            pass
-
-        def training_step(self, batch, batch_idx, optimizer_idx=None):
-            pass
-
-    trainer_options = dict(default_root_dir=tmpdir, max_epochs=1)
-    trainer = Trainer(**trainer_options)
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
 
     with pytest.raises(MisconfigurationException):
-        model = CurrentTestModel()
+        model = EvalModelTemplate(tutils.get_default_hparams())
+        model.configure_optimizers = LightningModule.configure_optimizers
```

Comment:
Why are you using LightningModule directly here?

Reply:
EvalModelTemplate has its own configure_optimizers

Reply:
ok, I ll add empty config here